runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/retries/delay/StandardRetryTokenBucket.kt

```diff
@@ -6,30 +6,31 @@
 package aws.smithy.kotlin.runtime.retries.delay
 
 import aws.smithy.kotlin.runtime.retries.policy.RetryErrorType
-import aws.smithy.kotlin.runtime.time.Clock
-import aws.smithy.kotlin.runtime.time.epochMilliseconds
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 import kotlin.math.ceil
 import kotlin.math.floor
 import kotlin.math.min
+import kotlin.time.ExperimentalTime
+import kotlin.time.TimeSource
 
 private const val MS_PER_S = 1_000
 
 /**
  * The standard implementation of a [RetryTokenBucket].
  * @param options The configuration to use for this bucket.
- * @param clock A clock to use for time calculations.
+ * @param timeSource A monotonic time source to use for calculating the temporal token fill of the bucket.
  */
-public class StandardRetryTokenBucket(
+@OptIn(ExperimentalTime::class)
```

Comment:
We've gotten ourselves in trouble in the past by using experimental stdlib features (https://github.com/awslabs/aws-sdk-kotlin/issues/622, https://github.com/awslabs/aws-sdk-kotlin/issues/736). I don't have a better suggestion here but something to consider w.r.t any versioning policy/strategy we define.

Reply:
Discussed offline and decided to keep usage of this type for now, despite it being marked as experimental. I've added a task [aws-sdk-kotlin#860](https://github.com/awslabs/aws-sdk-kotlin/issues/860) to track revisiting all usages of experimental/unstable APIs before GA.